File: 프로그래머스/2/49994. 방문 길이/방문 길이.py
def solution(dirs):
    # 상하우좌
    dx = [-1,1,0,0]
    dy = [0,0,1,-1]
    x,y = 0,0
    answer = 0
    visited = [[False] * 5 for _ in range(5)]
    vis = set()
    for i in range(len(dirs)):
        if dirs[i] == 'U':
            dr = 0
        elif dirs[i] == 'D':
            dr = 1
        elif dirs[i] == 'R':
            dr = 2
        elif dirs[i] == 'L':
            dr = 3
        nx = x + dx[dr]
        ny = y + dy[dr]
        if -5<=nx<=5 and -5<=ny<=5:
            vis.add((x,y,nx,ny))
            vis.add((nx,ny,x,y))
            x = nx
            y = ny
                
    return len(vis)//2

File: 프로그래머스/2/49994. 방문 길이/방문 길이.cpp
#include <string>
#include <unordered_map>
#include <set>
#include <iostream>
using namespace std;

int di[4] = {-1,1,0,0}, dj[4] = {0,0,1,-1};
unordered_map<char,int> dmap;
set<tuple<int, int, int, int>> vset;
string dr = "UDRL";
int solution(string dirs) {
    int answer = 0;
    int ci = 0, cj = 0;
    int ni,nj;
    char dir;
    // vset.insert({ci,cj});
    for (int i = 0; i < 4; i++){
        dmap[dr[i]] = i;
    }
    
    for (int i = 0; i < dirs.length();i++){
        dir = dirs[i];
        ni = ci+di[dmap[dir]]; 
        nj = cj+dj[dmap[dir]];
        cout << ci << " : "<< cj << " : " << answer << endl;
        // cout << ni << " : "<< nj << endl;

        if (-5<=ni && ni<=5 && -5<=nj && nj<=5) {
            // if (vset.find({ni,nj}) == vset.end())
            if (vset.find({ci,cj, ni, nj}) == vset.end()){
                answer++;
                vset.insert({ci,cj,ni,nj});
                vset.insert({ni,nj,ci,cj});
                }            
            ci = ni ; cj = nj;
            }
        }
    cout << ci << " : "<< cj << " : " << answer << endl;
    return answer;
}